File: LeetCode/0810. Chalkboard XOR Game/solution.cpp
/**
 * 168 / 168 test cases passed.
 * Runtime: 8 ms
 * Memory Usage: 12.5 MB 
 */
class Solution {
public:
    bool xorGame(vector<int>& nums) {
        if (nums.size() % 2 == 0) return true;
        int sum = 0;
        for (auto& num: nums) sum ^= num;
        return sum == 0;
    }
};

File: LeetCode/0810. Chalkboard XOR Game/solution.py
"""
168 / 168 test cases passed.
Runtime: 88 ms
Memory Usage: 15.1 MB
"""
class Solution:
    def xorGame(self, nums: List[int]) -> bool:
        return True if len(nums) % 2 == 0 else reduce(xor, nums) == 0
